serializable.ts

```diff
@@ -4,8 +4,21 @@ import { SerializePropertyOptionsMap } from "./serialize_property_options_map.ts
 import { defaultToJson } from "./to_json/default.ts";
 import { recursiveToJson } from "./to_json/recursive.ts";
 
+/** to be implemented by external authors on their models  */
+export declare interface TransformKey {
+  /** a function that will be called against
+   * every property key transforming the key
+   * with the provided function
+   */
+  tsTransformKey(key: string): string;
```

Comment:
It's possible to use a symbol here if we want maximum collision avoidance:
```ts
export const tsTransformKey: unique symbol = Symbol("tsTransformKey");

export declare interface TransformKey {
  [tsTransformKey](key: string): string;
}

export abstract class Serializable {
  public [tsTransformKey]?(key: string): string {
    return key;
  }
  ...
}

// Implementation
class TestTransformKey extends Serializable implements TransformKey {
  public [tsTransformKey](key: string): string {
    return `__${key}__`;
  }
}
```

This is probably way overkill though and would be annoying to use in practice

Reply:
i think we'd hit an issue with this when users go to implement their own they would also need to export the symbol and use it accordingly, I think its a bit much to ask of external authors.